provisioning/transport/amqp/src/AmqpTransportClient.cs

```diff
@@ -1,25 +1,84 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.Devices.Provisioning.Client;
+using System;
 using Microsoft.Azure.Devices.Shared;
 using System.Threading.Tasks;
 using System.Threading;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport.Http.Models;
 
-namespace Microsoft.Azure.Devices.Client.Transport
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Amqp
 {
     public class AmqpTransportClient : ProvisioningTransportClient
     {
+        public const int DefaultOperationPoolingIntervalMilliseconds = 2 * 1000;
+
         public TransportFallbackType FallbackType { get; private set; }
 
-        public AmqpTransportClient(TransportFallbackType transportFallbackType = TransportFallbackType.TcpWithWebSocketFallback)
+        public AmqpTransportClient(
+            TransportFallbackType transportFallbackType = TransportFallbackType.TcpWithWebSocketFallback)
         {
             FallbackType = transportFallbackType;
         }
 
-        public override Task<ProvisioningRegistrationResult> RegisterAsync(string globalDeviceEndpoint, string idScope, ProvisioningSecurityClient securityClient, CancellationToken cancellationToken)
+        public override async Task<ProvisioningRegistrationResult> RegisterAsync(
+            string globalDeviceEndpoint, 
+            string idScope, 
+            ProvisioningSecurityClient securityClient, 
+            CancellationToken cancellationToken)
         {
-            throw new System.NotImplementedException();
+            AmqpClientConnection client;
```

Comment:
I know this is work-in-progress but have you considered how multiplexing using a single client across multiple `ProvisioningClient`s would work?
Is it possible with the current structure? (I believe we may need an implicit or explicit OpenAsync that will create the first link?)

Reply:
Service side is not supporting multiplexing and there is no use case for now.  Multiplexing would not be possible for x509.